### PR TITLE
feat(inquest): mandatory comment, max / default rate

### PR DESCRIPTION
**Pull request by stonebuzz**
Add some options

- Max rate (default 5, max 5 like GLPI native behavior)
- Default rate (default 3 like GLPI native behavior)
- Mandatory comment if score is less than or equal to =  X  (0 -> not mandatory)

image

On save or update check if comment is required and add message after redirect

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: ajax/ticketsatisfaction.php [hunk 1]
```diff
@@ -98,6 +103,46 @@
...
+            ['value' => $inquest_default_rate,
+                'min'   => 1,
+                'max'   => 5,
+                'step'  => 1,
+                'unit'  => ''
+            ]
+        );
+        echo "</td></tr>";
+
+        echo "<tr class='tab_bg_1'><td width='50%'>" . __('Comment required if score is less than or equal to') . "</td>";
```
Comment (orthagh) [inline ajax/ticketsatisfaction.php:132]:
```suggestion
        echo "<tr class='tab_bg_1'><td width='50%'>" . __('Comment required if score is <= to') . "</td>";
```

maybe we can use a symbol instead a full wording here ?
your opinion ?

Comment (stonebuzz) [reply ajax/ticketsatisfaction.php:132]: I think most people will understand that

File: src/TicketSatisfaction.php [hunk 2]
```diff
@@ -123,7 +123,8 @@ public function showSatisactionForm($ticket)
 
            // Set default satisfaction to 3 if not set
             if (is_null($this->fields["satisfaction"])) {
-                $this->fields["satisfaction"] = 3;
+                $default_rate = Entity::getUsedConfig('inquest_default_rate', $ticket->fields['entities_id'], 'inquest_default_rate');
```
Comment (cedric-anne) [inline src/TicketSatisfaction.php:126]: `inquest_default_rate` inheritance should be based on `inquest_config` field.
```suggestion
                $default_rate = Entity::getUsedConfig('inquest_config', $ticket->fields['entities_id'], 'inquest_default_rate');
```

File: src/TicketSatisfaction.php [hunk 3]
```diff
@@ -132,7 +133,8 @@ public function showSatisactionForm($ticket)
 
             echo "<select id='satisfaction_data' name='satisfaction'>";
 
-            for ($i = 0; $i <= 5; $i++) {
+            $max_rate = Entity::getUsedConfig('inquest_max_rate', $ticket->fields['entities_id'], 'inquest_max_rate');
```
Comment (cedric-anne) [inline src/TicketSatisfaction.php:136]: Same remark.
```suggestion
            $max_rate = Entity::getUsedConfig('inquest_config', $ticket->fields['entities_id'], 'inquest_max_rate');
```

File: src/TicketSatisfaction.php [hunk 4]
```diff
@@ -171,10 +173,21 @@ public function showSatisactionForm($ticket)
 
     public function prepareInputForUpdate($input)
     {
+
         if ($input['satisfaction'] >= 0) {
             $input["date_answered"] = $_SESSION["glpi_currenttime"];
         }
 
+        $inquest_mandatory_comment = Entity::getUsedConfig('inquest_mandatory_comment', $input['entities_id'], 'inquest_mandatory_comment');
```
Comment (cedric-anne) [inline src/TicketSatisfaction.php:181]:
```suggestion
        $inquest_mandatory_comment = Entity::getUsedConfig('inquest_config', $input['entities_id'], 'inquest_mandatory_comment');
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**